Detectors/TRD/reconstruction/src/CruRawReader.cxx

```diff
@@ -623,27 +437,31 @@ int CruRawReader::processHalfCRU(int cruhbfstartoffset, int numberOfPreviousCRU,
     // let incrementErrors catch the undefined values of sector side stack and layer as if not set it will go so zero in the method, however if set, it means this is the second half cru header, and we have the values from the last one we read which
     // *SHOULD* be the same as this halfcruheader.
     incrementErrors(TRDParsingHalfCRUCorrupt);
-    mWordsRejected += mTotalHalfCRUDataLength32;
-    mHBFoffset32 += mTotalHalfCRUDataLength32; // go to the end of this halfcruheader and payload.
+    mWordsRejected += (mTotalHBFPayLoad / 4) - mHBFoffset32;
     return -2;
   }
 
   //get eventrecord for event we are looking at
   mIR.bc = mCurrentHalfCRUHeader.BunchCrossing; // correct mIR to have the physics trigger bunchcrossing *NOT* the heartbeat trigger bunch crossing.
-  //shift accordingly
-  mIR.bc = mCurrentHalfCRUHeader.BunchCrossing - o2::ctp::TriggerOffsetsParam::Instance().LM_L0;
-  if (mIR.bc < 0) {
```

Comment:
@bazinski I found the reason for the slightly different numbers: `mIR.bc` is unsigned, so the check you had hear never failed and we were not rejecting triggers which would be shifted to a negative bc. This is fixed with the changes in here